api/v1/common.go

```diff
@@ -0,0 +1,14 @@
+package v1
+
+// TODO add validation to ensure only git, image or local is specified
+type Container struct {
+	Git   *GitSource `json:"git,omitempty"`
+	Image string     `json:"image,omitempty"`
+}
+
+type GitSource struct {
+	URL string `json:"url"`
+	// Git refspec, for example, refs/heads/my-branch
+	Branch string `json:"branch,omitempty"`
```

Comment:
Should we have a default of `main` instead of `omitempty`?

```suggestion
	// kubebuilder:default:=main
	Branch string `json:"branch"`
```

Reply:
Actually, I bet if you omit the default branch will be selected regardless of if it's main or master

Reply:
That's correct. It will use the default branch configured on the remote so we shouldn't set it in substratus